tasks/libs/common/worktree.py

```diff
@@ -0,0 +1,109 @@
+"""Worktree utilities, used to execute tasks from this local repository (main) to a worktree with a different HEAD (e.g. 6.53.x).
+
+Common environment variables that can be used:
+- WORKTREE_NO_PULL: If set to any value, the worktree will not be pulled before running the command.
+"""
+
+import os
+from contextlib import contextmanager
+from pathlib import Path
+
+from invoke.exceptions import Exit
+
+from tasks.libs.common.color import Color, color_message
+from tasks.libs.common.git import get_current_branch
+
+WORKTREE_DIRECTORY = Path.cwd().parent / "datadog-agent-worktree"
+LOCAL_DIRECTORY = Path.cwd().resolve()
+
+
+def init_env(ctx, branch: str | None = None):
+    """Will prepare the environment for commands applying to a worktree.
+
+    To be used before each worktree section.
+    Will:
+    1. Add the agent worktree if not present.
+    2. Fetch the latest changes from the agent worktree.
+    """
+
+    if not WORKTREE_DIRECTORY.is_dir():
+        if not ctx.run(f"git worktree add '{WORKTREE_DIRECTORY}' origin/{branch or 'main'}", warn=True):
```

Comment:
What's the point of initialising the worktree to `main` by default? Should we make the branch a mandatory argument?

Reply:
In some cases, we want to enter the environment without changing the branch. `main` is used to create the worktree in any case. For example, some release tasks won't have the branch argument required such that we:
1. Switch to the target branch
2. Apply tasks such as `tag_modules` etc. without specifying again the branch